OpenRA.Game/Activities/Activity.cs

```diff
@@ -41,11 +41,13 @@ protected Activity ChildActivity
 		public Activity NextActivity { get; protected set; }
 
 		public bool IsInterruptible { get; protected set; }
+		public bool ChildHasPriority { get; protected set; }
 		public bool IsCanceling { get { return State == ActivityState.Canceling; } }
 
 		public Activity()
 		{
 			IsInterruptible = true;
+			ChildHasPriority = true;
```

Comment:
Having this as a flag doesn't feel right to me. Is there a specific reason you didn't implement this using a virtual method that the special cases can override?

Reply:
The idea is that there shouldn't be special cases: Either the child has priority over the parent or the parent has priority over the child. I've now refactored `Enter` to also fit into this paradigm (all other activities already did). This means that only the base `Activity` class needs to set `ChildActivity` directly and the setter can be made private.